### ISSUE TITLE
NullPointerException when parsing WITH .. INSERT .. SELECT without explicit INSERT column list

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11043]
This SQL causes an exception on the translate website:

```sql
with x (c) as (select 1)
insert into t -- Missing column list here
select c from x
```

The exception is:

```
java.lang.NullPointerException
	at org.jooq.impl.InsertImpl.values(InsertImpl.java:793)
	at org.jooq.impl.InsertImpl.values(InsertImpl.java:809)
	at org.jooq.impl.InsertImpl.values(InsertImpl.java:1)
	at org.jooq.impl.ParserImpl.parseInsert(ParserImpl.java:1995)
	at org.jooq.impl.ParserImpl.parseWith(ParserImpl.java:1105)
	at org.jooq.impl.ParserImpl.parseWith(ParserImpl.java:1055)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:1019)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:673)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:667)
```

**Comment by juriad** [comment]
I wanted to create a new issue but I guess it has very similar cause, so here we go:

The code bellow throw a NPE:
```
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Arrays$ArrayList.<init>(Arrays.java:4323)
	at java.base/java.util.Arrays.asList(Arrays.java:4310)
	at org.jooq.impl.InsertQueryImpl.setSelect(InsertQueryImpl.java:276)
	at org.jooq.impl.InsertImpl.select(InsertImpl.java:187)
	at org.jooq.impl.InsertImpl.select(InsertImpl.java:124)
```

It looks like `WithImpl` (https://github.com/jOOQ/jOOQ/blob/main/jOOQ/src/main/java/org/jooq/impl/WithImpl.java#L866) is the only place which calls the constructor without fields (`InsertImpl(Configuration configuration, WithImpl with, Table<R> into)` which leaves `fields` uninitialized and causes NPE later.

### Steps to reproduce the problem

Does not work:
```
with(name("a")).as(selectOne()).insertInto(table(name("b"))).select(selectFrom(name("a"))).getSQL();
```

Works (workaround):
```
with(name("a")).as(selectOne()).insertInto(table(name("b")), List.of()).select(selectFrom(name("a"))).getSQL();
```

Also works:
```
insertInto(table(name("b"))).select(selectFrom(name("a"))).getSQL()
```

Version jOOQ: 3.13.1 pro; I checked `main` here on Github and expect the same problem there.

**Comment by lukaseder** [comment]
Thanks a lot for your report, and for documenting the workaround here, @juriad.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.8 (#11529).
